File: 02/balance.py
def balance(start_balance, top_up, interest_rate):
    current_balance = start_balance + top_up
    return current_balance + start_balance * interest_rate/100

def calculate_balance(start_balance, top_up, interest_rate, n_years):
    # calculate balance for 1st year
    calculated_balance = balance(start_balance, top_up, interest_rate)

    for _ in range(2, n_years + 1):
        calculated_balance = balance(calculated_balance, top_up, interest_rate)

    return calculated_balance

if __name__ == '__main__':
    print(calculate_balance(1000, 200, 2, 3))

File: balance/balance.go
package balance

import (
	"fmt"
)

func calculateBalance(startBalance, topUpAmount, interestRate int) float32 {
	balance := startBalance + topUpAmount
	return float32(balance + startBalance*(interestRate/100))
}

func showSavedBalance(startBalance, topUpAmount, interestRate, numYears int) float32 {
	var accumulatedBalance float32
	for i := 1; i <= numYears; i++ {
		accumulatedBalance += calculateBalance(startBalance, topUpAmount, interestRate)
	}
	return accumulatedBalance
}

func main() {
	fmt.Println(showSavedBalance(1000, 200, 2, 2))
}
